### ISSUE TITLE
[quant] quant aware training mode.  result is different between "before convert"and "after convert to int8 model" when run the same image

### ISSUE THREAD

**Issue by zhangruiidol2013** [pytorch/pytorch#37747]
## To Reproduce

1.
1.
1.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by zhangruiidol2013** [comment]
For example:
    output0  = self.model(input_data)
    input_data = input_data.cpu()
    torch.backends.quantized.engine = 'fbgemm'
    quant_model = copy.deepcopy(self.model).cpu()
    torch.quantization.convert(quant_model.eval(), inplace=True)
    quant_model.eval()
    output1  = quant_model(input_data)
    
    output0 and output1 are different

**Comment by supriyar** [comment]
@zhangruiidol2013 are you comparing the raw output values? In that case there may be difference between float and quantized models. 
Please use the discussion forum for questions related to quantization. https://discuss.pytorch.org/c/quantization/17
